wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift

```diff
@@ -0,0 +1,240 @@
+        _ response: ZMTransportResponse,
+        action: CreateGroupConversationAction
+    ) {
+        var action = action
+
+        guard
+            let apiVersion = APIVersion(rawValue: response.apiVersion),
+            let rawData = response.rawData,
+            let payload = Payload.Conversation(rawData, apiVersion: apiVersion),
+            let newConversation = payload.updateOrCreate(in: context)
+        else {
+            Logging.network.warn("Can't process response, aborting.")
+            action.fail(with: .proccessingError)
+            return
+        }
+
+        switch newConversation.messageProtocol {
+        case .proteus:
+            context.saveOrRollback()
+            action.succeed(with: newConversation.objectID)
+
+        case .mls:
+            Logging.mls.info("created new conversation on backend, got group ID (\(String(describing: payload.mlsGroupID)))")
+
+            // Self user is creator, so we don't need to process a welcome message
+            newConversation.mlsStatus = .ready
+
+            guard let mlsService = context.mlsService else {
+                Logging.mls.warn("failed to create mls group: mlsService doesn't exist")
+                action.fail(with: .proccessingError)
+                return
+            }
+
+            guard let groupID = newConversation.mlsGroupID else {
+                Logging.mls.warn("failed to create mls group: conversation is missing group id.")
+                action.fail(with: .proccessingError)
+                return
+            }
+
+            do {
+                try mlsService.createGroup(for: groupID)
+            } catch let error {
+                Logging.mls.error("failed to create mls group: \(String(describing: error))")
+                action.fail(with: .proccessingError)
+                return
+            }
+
+            // If this is an mls conversation, then the initial participants won't have
+            // been added yet on the backend. This means that we must take the list of
+            // participants from the action instead of the local conversation.
+            let pendingParticipants = Set(action.qualifiedUserIDs).union(action.unqualifiedUserIDs.compactMap {
+                guard let localDomain = BackendInfo.domain else { return nil }
+                return QualifiedID(uuid: $0, domain: localDomain)
+            })
+
+            let selfUserID = ZMUser.selfUser(in: context).qualifiedID
+
+            let users = pendingParticipants.map { qualifiedID in
+                if qualifiedID == selfUserID {
+                    return MLSUser(qualifiedID, selfClientID: action.creatorClientID)
+                } else {
+                    return MLSUser(qualifiedID)
+                }
+            }
+
+            let reportSuccess = {
+                action.succeed(with: newConversation.objectID)
+            }
+
+            let reportFailure = {
+                action.fail(with: .proccessingError)
+            }
```

Comment:
what's the reason behind declaring this here instead of calling the action's methods directly in the task below ?